### DISCUSSION TITLE
chat

### DISCUSSION THREAD

**Discussion by rejetto** [rejetto/hfs#583] [General]
i was exchanging with a user @hvscan about a chat inside hfs.
At the moment such feature is not planned and outside the scope of the project, but there are still 2 ways to achieve that: as a plugin, or as an external service.

No such a plugin exists yet, and I'm not personally going to explore this possibility in the near future, but maybe later.

I've searched a bit for external services, and most things you find are 1-to-1 services to support customers/visitors.
I was searching instead a sort of chat room, where all visitors could chat without registering (welcome spam!). It's not super easy to find stuff like this, (also because hosting as a cost) and that's why I decided to share what i've found so far, that is https://chatango.com/
I don't like very much the look, but I guess you can add some css to improve things a bit.
For the rest, the experience of setting up was smooth as I desired, and I just had to paste a small html piece in HFS custom-html.
Of course the administrator has to register to chatango, but other people don't need to.

**Comment by PH5HP** [comment]
Hi,

Found another one at 

https://code.tutsplus.com/how-to-create-a-simple-web-based-chat-application--net-5931t

Kind regards,
Henk - PH5HP.NL - https://fileserver.ph5hp.nl

**Reply by PH5HP** [reply]
Hi,

Javascript based chat:

https://github.com/PubNubDevelopers/JavaScript-Simple-Chat

Ah ... I see you need an account somewhere...

Kind regards,
Henk - PH5HP.NL - https://fileserver.ph5hp.nl/

**Comment by rejetto** [comment]
i'm not considering any web chat here, but only those that can work with HFS

**Reply by PH5HP** [reply]
Hi Rejetto,

What does that mean? Is it possible to use the webserver of HFS with a chatbox? What are the requirements you need for a chatbox?

Kind regards,
Henk - PH5HP.NL - https://fileserver.ph5hp.nl/

**Reply by rejetto** [reply]
that means just that you try to make it work inside hfs :) If you succeed, well.
i got chatango working in hfs in 5-10 minutes, at first sight.
And it's not HFS making the requirements, it's the chat. The first chat you linked requires php, and HFS doesn't offer that.

**Reply by PH5HP** [reply]
Hi Rejetto,

Do you have your example running in a demo HFS server so I can see how it is implemented?

Kind regards,
Henk - ph5hp.nl - https://fileserver.ph5hp.nl/

**Reply by rejetto** [reply]
i don't have a demo hfs at the moment, as mine is private.
I just used their website to create the chat, and they gave me a small html code to paste in custom-html, as i said in my post.

**Reply by rejetto** [reply]
but i can just give you "my" code to paste
```html
<script id="cid0020000376344019355" data-cfasync="false" async src="//st.chatango.com/js/gz/emb.js" style="width: 200px;height: 300px;">{"handle":"hfs3","arch":"js","styles":{"a":"CC0000","b":100,"c":"FFFFFF","d":"FFFFFF","k":"CC0000","l":"CC0000","m":"CC0000","n":"FFFFFF","p":"10","q":"CC0000","r":100,"pos":"br","cv":1,"cvbg":"CC0000","cvw":75,"cvh":30}}</script>
```

**Reply by PH5HP** [reply]
Hi Rejetto,

TNX... I have now an idea how it looks like.

 I signed up with Chatango for now, but still looking for a chat without needing an external service.

Kind regards,
Henk - ph5hp.nl - https://fileserver.ph5hp.nl/

**Comment by damienzonly** [comment]
Hi everyone, we just released a chat plugin! go check it out https://github.com/rejetto/hfs/discussions/641
